wikidataintegrator/wdi_helpers/publication.py

```diff
@@ -505,11 +509,47 @@ def arxiv_api_to_publication(ext_id, id_type='arxiv'):
     return publication
 
 
+def biorxiv_api_to_publication(ext_id, id_type='biorxiv'):
+    """Make a :class:`Publication` from a bioRxiv identifier."""
+    url = f'https://api.biorxiv.org/details/biorxiv/10.1101/{ext_id}'
+    headers = {
+        'User-Agent': config['USER_AGENT_DEFAULT']
+    }
+    response = requests.get(url, headers=headers)
+    response.raise_for_status()
+    revisions = response.json()['collection']
+    latest_revision = revisions[-1]
+    version = len(revisions)
+    authors = [author.strip() for author in latest_revision['authors'].split(';')]
+
+    publication = Publication(
+        title=latest_revision['title'],
+        ref_url=f'https://www.biorxiv.org/content/10.1101/{ext_id}v{version}',
+        authors=[
+            {
+                'full_name': author,
+            }
+            for author in authors
+            if author
+        ],
+        publication_date=datetime.datetime.strptime(latest_revision['date'], '%Y-%m-%d'),
+        ids={
+            'biorxiv': ext_id,
+            'doi': latest_revision['doi'],
```

Comment:
This is the bioRxiv DOI. Later, when the paper is accepted, it links the actual published DOI in a different field, `latest_revision['published']` field.

I'm not sure how this would get handled - are some fields updated if they change in the Publication data model? In that case, we could assign it this way:

````suggestion
            'doi': latest_revision.get('publication', latest_revision['doi']),
```

Reply:
@andrawaag any thoughts on this? what are the implications of adding a DOI pointing to a preprint rather than a peer reviewed article?

I think right now the same wikidata item encompasses both a preprint and a postprint (i don't agree with this, but that's what I gathered from the scholia group). How should we update the item when it gets accepted?

Reply:
@cthoyt I find this a tough question. Personally, I am a bit ambivalent on adding preprints to Wikidata at all. What I fear is that "stamp-collecting" of preprints could severely affect the performance of Wikidata. Like with the regular citations, a lot of items have been added just because we can, not because the items are used as reference or evidence in Wikidata, leading to many unconnected nodes. I am more in favour of adding citation only if they are actually being used/connected by other nodes in the Wikidata graph. With this in mind, I can see some merit in merging both the pre and post prints ones a paper gets successfully through the peer-review process. However, at the same time the preprint on its own can come with different versions, and such a provenance has value to exist in separate items. But given the already sheer size of Wikidata, I wonder if this should not be handled in a separate Wikibase. 

However, I think that resolution in this issue, should happen on the Wikicite platforms.

Reply:
Thanks for the response. Maybe a separate service should check for and make these kinds of updates.